libs/designer-ui/src/lib/createConnection/index.tsx

```diff
@@ -185,8 +186,9 @@ export const CreateConnection = (props: CreateConnectionProps): JSX.Element => {
         return isServicePrinicipalConnectionParameter(key) && isServicePrincipalParameterVisible(key, parameter);
       if (legacyManagedIdentitySelected) return false; // TODO: Riley - Only show the managed identity parameters (which is none for now)
       if (constraints?.hidden === 'true' || constraints?.hideInUI === 'true') return false;
-      const dependencyParam = constraints?.dependentParameter;
-      if (dependencyParam && parameterValues[dependencyParam.parameter] !== dependencyParam.value) return false;
+      const dependentParam = constraints?.dependentParameter;
+      const currentDependencyValue = getCaseInsensitiveProperty(parameterValues, dependentParam?.parameter);
```

Comment:
we have a function in functions.ts called "getPropertyValue", think we can use that here instead